agent/logger/log.go

```diff
@@ -55,31 +55,40 @@ var Config *logConfig
 func logfmtFormatter(params string) seelog.FormatterFunc {
 	return func(message string, level seelog.LogLevel, context seelog.LogContextInterface) interface{} {
 		cc, ok := context.CustomContext().(map[string]string)
-		var customContext string
-		if ok && len(cc) > 0 {
-			var sortedContext []string
-			for k, v := range cc {
-				sortedContext = append(sortedContext, k+"="+v)
-			}
-			sort.Strings(sortedContext)
-			customContext = " " + strings.Join(sortedContext, " ")
+		if !ok {
+			cc = map[string]string{}
 		}
-		return fmt.Sprintf(`level=%s time=%s msg=%q module=%s%s
-`, level.String(), context.CallTime().UTC().Format(time.RFC3339), message, context.FileName(), customContext)
+		if _, ok = cc["module"]; !ok {
+			cc["module"] = context.FileName()
+		}
+
+		var ccStr string
+		var ccSorted []string
+		for k, v := range cc {
+			ccSorted = append(ccSorted, k+"="+v)
+		}
+		sort.Strings(ccSorted)
+		ccStr = " " + strings.Join(ccSorted, " ")
+		return fmt.Sprintf(`level=%s time=%s msg=%q%s
+`, level.String(), context.CallTime().UTC().Format(time.RFC3339), message, ccStr)
```

Comment:
could this and lines 80-91 be a single function with a `contextSort=True` flag?

Reply:
I've refactored out the reusable code from here. Didn't make the sorting part reusable because that only applies to the logfmt formatter and not the JSON. That part would have been messy to factor out because they require formatting the 'context' map differently to match the desired output.